### ISSUE TITLE
Discussion: Logging

### ISSUE THREAD

**Issue by maede97** [cevi/automatic_walk-time_tables#30]
As mentioned in #22 we should further improve the logging:

> The main aspect of this issue is that we import the module with the other loggers before parsing the arguments (namely at the top of the main file)
> Consistently use classes among all files using logger, then instantiate the logger by calling logging.getLogger(__name__) in the constructor of the class.
> Slacker variant: inline the import of as many as possible modules to a function called once the args are parsed.
> Version (2) should work but is against the philosophy of python to put import statements at the top of a file.

**Comment by maede97** [comment]
I think this should be done via the first option. Adding Classes would (in my opinion) make much sense and simplify the structure more.

**Comment by maede97** [comment]
@wp99cp If you agree, we should potentially open a new issue to track the changes for the individual classes.

**Comment by wp99cp** [comment]
> I think this should be done via the first option. Adding Classes would (in my opinion) make much sense and simplify the structure more.

Using classes is also useful for #29, since (if properly implemented) it will allow for multiple exports in parallel without any interference (currently a known issue of #29).

**Comment by maede97** [comment]
> > I think this should be done via the first option. Adding Classes would (in my opinion) make much sense and simplify the structure more.
> 
> Using classes is also useful for #29, since (if properly implemented) it will allow for multiple exports in parallel without any interference (currently a known issue of #29).

Good point!

**Comment by maede97** [comment]
So I guess we :
- should implement the class-structure first, then merge back into master, merge master into #29 and then continue there?
- or finalize the web-interface and use classes later as a enhancement

**Comment by wp99cp** [comment]
I prefer variant (1), as it allows parallel development. If you look at the changes of #29, you'll see that the implementation of the webserver is done as a separate layer on top of the current implementation, therefore only minor changes are needed in the existing codebase. 
> * should implement the class-structure first, then merge back into master, merge master into [add basic webinterface #29](https://github.com/cevi/automatic_walk-time_tables/pull/29) and then continue there?

**Comment by wp99cp** [comment]
76c332ac3735b94c13c0163144ff9618b8787c8a in #32 will add a command line argument for the logger level.  It uses a mixture of methods (1) and (2) as discussed in https://github.com/cevi/automatic_walk-time_tables/pull/28#issuecomment-998655209. I.g. I created a global AutomatedWalkTableGenerator class and lazy imports to configure the logger.
